atoma-proxy-service/src/handlers/auth.rs

```diff
@@ -559,3 +522,67 @@ pub(crate) async fn get_balance(
             })?,
     ))
 }
+
+/// OpenAPI documentation for the get_user_profile endpoint.
+///
+/// This struct is used to generate OpenAPI documentation for the get_user_profile
+/// endpoint. It uses the `utoipa` crate's derive macro to automatically generate
+/// the OpenAPI specification from the code.
+#[derive(OpenApi)]
+#[openapi(paths(get_user_profile))]
+pub(crate) struct GetUserProfile;
+
+/// Retrieves the user profile for the user.
+///
+/// # Arguments
+///
+/// * `proxy_service_state` - The shared state containing the state manager
+/// * `headers` - The headers of the request
+///
+/// # Returns
+///
+/// * `Result<Json<Value>>` - A JSON response containing the user profile
+///
+/// # Errors
+///
+/// * If the user ID cannot be retrieved from the token, returns a 401 Unauthorized error
+/// * If the user profile cannot be retrieved, returns a 500 Internal Server Error
+#[utoipa::path(
+    get,
+    path = "",
+    security(
+        ("bearerAuth" = [])
+    ),
+    responses(
+        (status = OK, description = "Retrieves the user profile for the user"),
+        (status = UNAUTHORIZED, description = "Unauthorized request"),
+        (status = INTERNAL_SERVER_ERROR, description = "Failed to get user profile")
+    )
+)]
+#[instrument(level = "info", skip_all)]
+pub(crate) async fn get_user_profile(
+    State(proxy_service_state): State<ProxyServiceState>,
+    headers: HeaderMap,
+) -> Result<Json<UserProfile>> {
+    let jwt = get_jwt_from_headers(&headers)?;
+
+    let user_id = proxy_service_state
+        .auth
+        .get_user_id_from_token(jwt)
+        .await
+        .map_err(|e| {
+            error!("Failed to get user ID from token: {:?}", e);
+            StatusCode::UNAUTHORIZED
```

Comment:
Use our custom error type, it provides better messaging

Reply:
we do not have custom error on axum server